services/validators/envValidator.go

```diff
@@ -151,6 +151,60 @@ func closeDBConnection(handle *sql.DB) {
 	}
 }
 
+func killDanglingDBConnections(db *sql.DB) {
+
+	rows, err := db.Query(`SELECT PID, QUERY_START, WAIT_EVENT_TYPE, WAIT_EVENT, STATE, QUERY
+							FROM PG_STAT_ACTIVITY
+							WHERE PID <> PG_BACKEND_PID()
+							AND APPLICATION_NAME = (
+								SELECT APPLICATION_NAME FROM PG_STAT_ACTIVITY WHERE PID = PG_BACKEND_PID()
+							)`)
+
+	if err != nil {
+		panic(fmt.Errorf("error occurred with querying pg_stat_activity table for finding dangling connections: %v", err.Error()))
+	}
+	defer rows.Close()
+
+	type danglingConnRowT struct {
+		pid           int
+		queryStart    string
+		waitEventType string
+		waitEvent     string
+		state         string
+		query         string
+	}
+	type danglingConnsT []*danglingConnRowT
+
+	dangling := danglingConnsT(make([]*danglingConnRowT, 0))
+	for rows.Next() {
+		var row danglingConnRowT = danglingConnRowT{}
+		err := rows.Scan(&row.pid, &row.queryStart, &row.waitEventType, &row.waitEvent, &row.state, &row.query)
+		if err != nil {
+			panic(err)
+		}
+		dangling = append(dangling, &row)
+	}
+
+	if len(dangling) > 0 {
+		pkgLogger.Warn(fmt.Sprintf("Killing %d dangling connection(s)", len(dangling)))
+		for i, rowPtr := range dangling {
+			pkgLogger.Warn(fmt.Sprintf("dangling connection #%d: %+v", i+1, *rowPtr))
+		}
+
+		_, err := db.Exec(`SELECT PG_TERMINATE_BACKEND(PID)
+							FROM PG_STAT_ACTIVITY
+							WHERE APPLICATION_NAME = (
+								SELECT APPLICATION_NAME FROM PG_STAT_ACTIVITY WHERE PID = PG_BACKEND_PID()
+							)
+							AND PID <> PG_BACKEND_PID();`)
+
+		if err != nil {
+			panic(fmt.Errorf("error occurred with executing pg_terminate_backend for dangling connections: %v", err.Error()))
```

Comment:
Not sure if we should panic here. Even if we failed to terminate some of these connections, shouldn't we give the process a chance to run?

Reply:
Would it make sense to the let the process run in a stuck state where queries are blocked by these dangling connections? I personally would prefer a panic here instead if we are unable to kill all the dangling connections

Reply:
IMO, an error here could only imply a malfunctioning database, thus I would argue that it doesn't make sense to resume with the server's bootstrap.